lock-api-objects/src/main/java/com/palantir/lock/watch/LockMetadata.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.lock.watch.LockMetadata.Created;
+import com.palantir.lock.watch.LockMetadata.Deleted;
+import com.palantir.lock.watch.LockMetadata.Unchanged;
+import com.palantir.lock.watch.LockMetadata.Updated;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Immutable;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = Unchanged.class, name = Unchanged.TYPE),
+    @JsonSubTypes.Type(value = Updated.class, name = Updated.TYPE),
+    @JsonSubTypes.Type(value = Deleted.class, name = Deleted.TYPE),
+    @JsonSubTypes.Type(value = Created.class, name = Created.TYPE),
+})
+public interface LockMetadata {
+    <T> T accept(Visitor<T> visitor);
+
+    static Unchanged unchanged() {
+        return ImmutableUnchanged.builder().build();
```

Comment:
We could use a singleton here, but we might want to add the new value later anyways for validation purposes